database/misc/mongodb_replset.py

```diff
@@ -0,0 +1,218 @@
+    from pymongo.read_preferences import ReadPreference
+    from pymongo.uri_parser import split_hosts
+except ImportError:
+    pymongo = None
+
+
+def load_mongocnf_creds():
+    """ Read credentials from ~/.mongodb.cnf file, when exists. """
+    config = ConfigParser.RawConfigParser()
+    mongocnf = os.path.expanduser('~/.mongodb.cnf')
+
+    try:
+        config.readfp(open(mongocnf))
+        creds = {
+            'user': config.get('client', 'user'),
+            'password': config.get('client', 'pass')
+        }
+    except (ConfigParser.NoOptionError, IOError):
+        return False
+
+    return creds
+
+
+def replset_conf(client):
+    """ Return replica set configuration; the same as rs.conf() in shell.
+
+    :param client: initialized Mongo client
+    :return: replica set config
+    """
+    return client['local'].system.replset.find_one()
+
+
+def add_members(client, members):
+    """ Add new members to the replica set.
+
+    :param client: initialized Mongo client
+    :param members: list of tuples that defines hostnames and ports of the
+                    replica set members;
+                    example: `[(mango0, 27017), (mango1, 27018)]`
+    """
+    conf = replset_conf(client)
+    curr_hosts = [m['host'] for m in conf['members']]
+    new_hosts = [join_colon(x) for x in members]
+
+    new_id = max([int(x['_id']) for x in conf['members']]) + 1
+    for host in (set(new_hosts) - set(curr_hosts)):
+        conf['members'] += [{'_id': new_id, 'host': host}]
+        new_id += 1
+
+    conf['version'] += 1
+    client.admin.command('replSetReconfig', conf)
+
+
+def replset_initiate(client, name, members):
+    """ Initiate replica set with the specified members.
+
+    :param client: initialized Mongo client
+    :param name: name of the replica set to initiate
+    :param members: list of tuples that defines hostnames and ports of the
+                    replica set members;
+                    example: `[(mango0, 27017), (mango1, 27018)]`
+    """
+    hosts = [{'_id': idx, 'host': join_colon(val)} for idx, val in enumerate(members)]
+    conf = {'_id': name, 'members': hosts}
+    client.admin.command('replSetInitiate', conf)
+
+
+def join_colon(iterable):
+    return ':'.join(str(s) for s in iterable)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec={
+            'login_user':     {'aliases': ['user']},
+            'login_password': {'aliases': ['password']},
+            'login_host':     {'aliases': ['host'], 'default': 'localhost'},
+            'login_port':     {'aliases': ['port'], 'default': 27017},
+            'hosts':          {'aliases': ['members'], 'required': True},
+            'replica_set':    {'aliases': ['replset'], 'required': True}
+        }
+    )
+
+    if not pymongo:
+        module.fail_json(msg='Python module "pymongo" must be installed.')
+
+    user, password, host, port, hosts, replset = (
```

Comment:
Is this necessary?  Seems overly complicated and too "clever".  I'd advocate for something more explicit if this is really needed.